openqa-advanced-retrigger-jobs

```diff
@@ -1,16 +1,105 @@
+--additional-filters \"test not like '%%:investigate:%%'\" --dry-run
+"""
+
+import argparse
+import logging
+import subprocess
+import sys
+from datetime import datetime
+
+logging.basicConfig()
+log = logging.getLogger(sys.argv[0] if __name__ == "__main__" else __name__)
+
+
+class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter, argparse.RawDescriptionHelpFormatter):
+    """Preserve multi-line __doc__ and provide default arguments in help strings."""
+
+    pass
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description=__doc__, formatter_class=CustomFormatter)
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        help="Increase verbosity level, specify multiple times to increase verbosity",
+        action="count",
+        default=1,
+    )
+    parser.add_argument("-H", "--host", default="openqa.opensuse.org", help="Target openQA host")
+    parser.add_argument(
+        "-s",
+        "--failed-since",
+        default=datetime.today().isoformat(),
+        help="Filter jobs failed since this date",
+    )
+    parser.add_argument("-w", "--worker", default=None, help="Filter jobs assigned to this worker")
+    parser.add_argument("-i", "--instance", default=None, help="Instance of the worker")
+    parser.add_argument("-r", "--result", default="incomplete", help="Filter jobs with this result")
+    parser.add_argument(
+        "-a",
+        "--additional-filters",
+        default=None,
+        help="Additional filters for the SQL query",
+    )
+    parser.add_argument("-c", "--comment", default=None, help="Comment to add to the retriggered jobs")
+    parser.add_argument(
+        "-d",
+        "--dry-run",
+        action="store_true",
+        help="If set, only print the actions without executing",
+    )
+    args = parser.parse_args()
+    verbose_to_log = {
+        0: logging.CRITICAL,
+        1: logging.ERROR,
+        2: logging.WARN,
+        3: logging.INFO,
+        4: logging.DEBUG,
+    }
+    logging_level = logging.DEBUG if args.verbose > 4 else verbose_to_log[args.verbose]
+    log.setLevel(logging_level)
+    return args
+
+def post(dry_run: bool, host: str, route: str, *args):
+    cmd=('openqa-cli', 'api', '--host', host, '-X', 'POST', route, *args)
+    if dry_run:
+        return print(f"dry run: {cmd}")
+    subprocess.run(cmd)
+
+def main(args):
+    log.debug(args)
+    instance_string = f" and instance='{args.instance}'" if args.instance else ""
+    worker_string = f"assigned_worker_id in (select id from workers where (host='{args.worker}'{instance_string})) and " if args.worker else ""
+    additional_filters = f" and {args.additional_filters}" if args.additional_filters else ""
+
+    query = (
+        f"select id from jobs where ({worker_string}result='{args.result}' "
+        f"and clone_id is null and t_finished >= '{args.failed_since}'{additional_filters});"
+    )
```

Comment:
Normally I'd say this needs escaping but of course the previous script also didn't have that…

Reply:
I don't know what additional escaping you mean. Why do you mean we *need* escaping?

Reply:
I think he means something in the lines of https://www.psycopg.org/psycopg3/docs/basic/params.html#execute-arguments

Currently e.g. `args.result` could be used to insert arbitrary SQL statements

Reply:
While malice is very unlikely (given you'd need access anyway) it could help if people accidentally pass arguments which cause the query to break

Reply:
And one can also (accidentally) break out of the whole `psql` invocation.